doc/decisions/0_drafts/transformations.md

```diff
@@ -0,0 +1,152 @@
+# Transformations
+
+## Problem
+
+Certain plugins, e.g. `rename`, perform extensive transformations on keys and keysets within Elektra.
+Those transformations include, but are not limited to:
+
+- Changing the names of the keys inbetween `kdbGet` and `kdbSet` operations
+- Changing values back and forth for normalization, e.g. `true` -> `1`, `1` -> `true`
+- Adding and removing metadata
+
+While these features are useful, they do create feature-interaction problems.
+More specifically, problems have been observed in conjunction with the following (overlapping) types of plugins:
+
+- notification plugins
+- plugins that do change tracking
+
+The problem, in general, can be described as: Which phase of the KDB should be used for notifications/change tracking?
```

Comment:
I don't understand this sentence. It also seems to me to be more specific and not more general?

Reply:
I think "representation" is the better word for "phase" here.